ravenframework/Models/PostProcessors/Validations/Representativity.py

```diff
@@ -0,0 +1,491 @@
+    _, notFound = paramInput.findNodesAndExtractValues(['prototypeParameters',
+                                                               'targetParameters'])
+    # notFound must be empty
+    assert(not notFound)
+
+  def run(self, inputIn):
+    """
+      This method executes the postprocessor action. In this case it computes representativity/bias factors, corrected data, etc.
+
+      @ In, inputIn, dictionary of data to process
+      @ Out, evaluation, dict, dictionary containing the post-processed results
+    """
+    dataSets = [data for _, _, data in inputIn['Data']]
+    pivotParameter = self.pivotParameter
+    names=[]
+    if isinstance(inputIn['Data'][0][-1], xr.Dataset):
+      names = [self.getDataSetName(inp[-1]) for inp in inputIn['Data']]
+      if len(inputIn['Data'][0][-1].indexes) > 1 and self.pivotParameter is None:
+        if 'dynamic' not in self.dynamicType: #self.model.dataType:
+          self.raiseAnError(IOError, "The validation algorithm '{}' is not a dynamic model but time-dependent data has been inputted in object {}".format(self._type, inputIn['Data'][0][-1].name))
+        else:
+          pivotParameter = self.pivotParameter
+    evaluation ={k: np.atleast_1d(val) for k, val in  self._evaluate(dataSets, **{'dataobjectNames': names}).items()}
+
+    ## TODO: This is a placeholder to remember the time dependent case
+    # if pivotParameter:
+    #   # Uncomment this to cause crash: print(dataSets[0], pivotParameter)
+    #   if len(dataSets[0][pivotParameter]) != len(list(evaluation.values())[0]):
+    #     self.raiseAnError(RuntimeError, "The pivotParameter value '{}' has size '{}' and validation output has size '{}'".format( len(dataSets[0][self.pivotParameter]), len(evaluation.values()[0])))
+    #   if pivotParameter not in evaluation:
+    #     evaluation[pivotParameter] = dataSets[0][pivotParameter]
+    return evaluation
+
+  def _evaluate(self, datasets, **kwargs):
+    """
+      Main method to "do what you do".
+      @ In, datasets, list, list of datasets (data1,data2,etc.) to used.
+      @ In, kwargs, dict, keyword arguments
+      @ Out, outputDict, dict, dictionary containing the results {"feat"_"target"_"metric_name":value}
+    """
+    # # ## Analysis:
+    # # 1. Compute mean and variance:
+    # For mock model
+    self._computeMoments(datasets[0], self.prototypeParameters, self.prototypeOutputs)
+    measurableNames = [s.split("|")[-1] for s in self.prototypeOutputs]
+    measurables = [datasets[0][var].meanValue for var in measurableNames]
+    # For target model
+    self._computeMoments(datasets[1], self.targetParameters, self.targetOutputs)
+    FOMNames = [s.split("|")[-1] for s in self.targetOutputs]
+    FOMs = np.atleast_2d([datasets[1][var].meanValue for var in FOMNames]).reshape(-1,1)
+    # # 2. Propagate error from parameters to experiment and target outputs.
+    # For mock model
+    self._computeErrors(datasets[0],self.prototypeParameters, self.prototypeOutputs)
```

Comment:
personally, I think the way you are using method "computeMoments and computeErrors" is more like C-type. datasets is passed as reference, and updated during the method calculation. I feel like we need to have a discussion on these two methods. I think @joshua-cogliati-inl will have a better comments for these two functions.

Reply:
@joshua-cogliati-inl, @wangcj05, do you have specific suggestions o make this more efficient?

Reply:
@joshua-cogliati-inl, any suggestions/insights here?

Reply:
Hm, `_computeMoments` should probably return the data calculated, instead of updating the dataset.
As in something like:
```python
     measurables = self._computeMoments(datasets[0], self.prototypeParameters, self.prototypeOutputs)
...
    FOMs = self._computeMoments(datasets[1], self.targetParameters, self.targetOutputs)
```
Similarly I would expect `_computeErrors` to return data instead of modifying the dataset.  As in it seems strange to me that the postprocessor is changing the dataset, but then also returning data in `out`

That said, there are probably other ways to do this.